doc/interpolation.rst

```diff
@@ -0,0 +1,260 @@
+
+array-like coordinates can be also accepted
+
+.. ipython:: python
+
+    # label lookup
+    da.sel(time=[2, 3], space=[0.1, 0.2])
+
+    # interpolation
+    da.interp(time=[1.5, 2.5], space=[0.15, 0.25])
+
+
+Interpolation methods
+---------------------
+
+We use scipy.interpolate.interp1d for 1-dimensional interpolation and
+scipy.interpolate.interpn for multi-dimensional interpolation.
+
+The interpolation method can be specified by the ``method`` optional argument.
+
+.. ipython:: python
+
+    da = xr.DataArray(np.sin(np.linspace(0, 2 * np.pi, 10)), dims='x',
+                      coords={'x': np.linspace(0, 1, 10)})
+
+    da.plot.line('o', label='original')
+    da.interp(x=np.linspace(0, 1, 100)).plot.line(label='linear (default)')
+    da.interp(x=np.linspace(0, 1, 100), method='cubic').plot.line(label='cubic')
+    @savefig interpolation_sample1.png width=4in
+    plt.legend()
+
+Additional keyword arguments can be passed to scipy's functions.
+
+.. ipython:: python
+
+    # fill 0 for the outside of the original coordinates.
+    da.interp(x=np.linspace(-0.5, 1.5, 10), kwargs={'fill_value': 0.0})
+    # extrapolation
+    da.interp(x=np.linspace(-0.5, 1.5, 10), kwargs={'fill_value': 'extrapolate'})
+
+
+Advanced Interpolation
+----------------------
+
+:py:meth:`~xarray.DataArray.interp` accepts :py:class:`~xarray.DataArray`
+as similar to :py:meth:`~xarray.DataArray.sel`, which enables us more advanced interpolation.
+Based on the dimension of the new coordinate passed to :py:meth:`~xarray.DataArray.interp`, the dimension of the result are determined.
+
+For example, if you want to interpolate a two dimensional array along a particular dimension, as illustrated below,
+you can pass two 1-dimensional :py:class:`~xarray.DataArray` s with
+a common dimension as new coordinate.
+
+.. image:: _static/advanced_selection_interpolation.svg
+    :height: 200px
+    :width: 400 px
+    :alt: advanced indexing and interpolation
+    :align: center
+
+For example:
+
+.. ipython:: python
+
+    da = xr.DataArray(np.sin(0.3 * np.arange(20).reshape(5, 4)),
+                      [('x', np.arange(5)),
+                       ('y', [0.1, 0.2, 0.3, 0.4])])
+    # advanced indexing
+    x = xr.DataArray([0, 2, 4], dims='z')
+    y = xr.DataArray([0.1, 0.2, 0.3], dims='z')
+    da.sel(x=x, y=y)
+
+    # advanced interpolation
+    x = xr.DataArray([0.5, 1.5, 2.5], dims='z')
+    y = xr.DataArray([0.15, 0.25, 0.35], dims='z')
+    da.interp(x=x, y=y)
+
+where values on the original coordinates
+``(x, y) = ((0.5, 0.15), (1.5, 0.25), (2.5, 0.35))`` are obtained by the
+2-dimensional interpolation and mapped along a new dimension ``z``.
+
+If you want to add a coordinate to the new dimension ``z``, you can supply
+:py:class:`~xarray.DataArray` s with a coordinate,
+
+.. ipython:: python
+
+    x = xr.DataArray([0.5, 1.5, 2.5], dims='z', coords={'z': ['a', 'b','c']})
+    y = xr.DataArray([0.15, 0.25, 0.35], dims='z')
```

Comment:
for clarity, would it make sense to specify the z-coordinate for both x and y here?